benchopt/base.py

```diff
@@ -469,7 +484,9 @@ def __call__(self, result):
     def set_dataset(self, dataset):
         self._dataset = dataset
         assert self.is_installed(raise_on_not_installed=True)
-        data = dataset._get_data()
+
+        uses_seed = class_uses_seeding(dataset)
+        data = dataset._get_data(uses_seed)
```

Comment:
This should be directly handled in the dataset, not clear why it needs to be done in the objective?